src/components/common/SceneWrapper.tsx

```diff
@@ -372,3 +375,38 @@ const SceneFooter = styled(View)({
   left: 0,
   right: 0
 })
+
+/**
+ * A hacky component to fix floating nav-bar issues caused by the keyboard being
+ * opened before a navigation is dispatched.
+ *
+ * This component listens for the 'beforeRemove' event on the navigation object
+ * and prevents the default behavior of leaving the screen if the keyboard is
+ * visible. It dismisses the keyboard and waits for the 'keyboardWillHide'
+ * event before dispatching the navigation action.
+ *
+ * @returns null
+ */
+function FloatingNavFixer() {
+  const navigation = useNavigation()
+
+  React.useEffect(() => {
+    const unsubscribe = navigation.addListener('beforeRemove', e => {
+      if (Keyboard.isVisible()) {
+        // Prevent default behavior of leaving the screen
+        e.preventDefault()
+
+        Keyboard.dismiss()
+
+        const keyboardDidHideListener = Keyboard.addListener('keyboardWillHide', () => {
+          navigation.dispatch(e.data.action)
+          keyboardDidHideListener.remove()
+        })
```

Comment:
Put the `Keyboard.dismiss()` call *after* the subscription, so there's no way we miss the event.